judoscale-good_job/lib/judoscale/good_job/metrics_collector.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "judoscale/job_metrics_collector"
+require "judoscale/job_metrics_collector/active_record_helper"
+require "judoscale/metric"
+
+module Judoscale
+  module GoodJob
+    class MetricsCollector < Judoscale::JobMetricsCollector
+      include ActiveRecordHelper
+
+      def self.adapter_config
+        Judoscale::Config.instance.good_job
+      end
+
+      def collect
+        metrics = []
+        time = Time.now.utc
+
+        all_queues = ::GoodJob::JobsFilter.new({}).queues.keys
+        self.queues |= all_queues
+
+        # logically we don't need the finished_at condition, but it lets postgres use the indexes
+        oldest_execution_time_by_queue = run_silently do
+          ::GoodJob::Execution
+            .where(performed_at: nil, finished_at: nil)
+            .group(:queue_name)
+            .pluck(:queue_name, Arel.sql("min(coalesce(scheduled_at, created_at))"))
+            .to_h
+        end
+
+        if track_busy_jobs?
+          busy_count_by_queue = run_silently do
+            ::GoodJob::Execution.running.group(:queue_name).count
```

Comment:
@bensheldon One of Judoscale's features is tracking "busy jobs" so users can optionally suppress downscaling when jobs are actively running. This avoids terminating long-running jobs that can't be cleanly restarted. (Not an ideal situation, but one in which a lot of teams find themselves.)

Does the query above look correct to you for obtaining a count of actively running jobs grouped by queue?

Reply:
Yep, that query looks good. `GoodJob::Execution.running` will be incorrect if the process is SIGKILLed because it doesn't look at Advisory Locks, though that also makes it a much lighter-weight query. I think it's appropriate.

(The alternative would be `GoodJob::Job.running`)

Sorry for the slow reply on this. Happy to see it merged 🙌🏻

Reply:
Thanks!